core/src/validation.jl

```diff
@@ -12,12 +12,13 @@ neighbortypes(::Val{:basin}) = Set((
     :outlet,
     :user_demand,
 ))
-neighbortypes(::Val{:terminal}) = Set{Symbol}() # only endnode
+neighbortypes(::Val{:terminal}) = Set{Symbol}()
+neighbortypes(::Val{:junction}) = Set((:basin, :junction))
```

Comment:
This looks like the original directed Confluence validation of https://github.com/Deltares/Ribasim/issues/1609#issuecomment-2730054537.

But Junction should also be able to go to connector nodes right?